src/jit/hwintrinsicxarch.cpp

```diff
@@ -69,19 +70,59 @@ NamedIntrinsic HWIntrinsicInfo::lookupId(const char* className, const char* meth
     return NI_Illegal;
 }
 
+//------------------------------------------------------------------------
+// X64VersionOfIsa: Gets the corresponding 64-bit only InstructionSet for a given InstructionSet
+//
+// Arguments:
+//    isa -- The InstructionSet ID
+//
+// Return Value:
+//    The 64-bit only InstructionSet associated with isa
+static InstructionSet X64VersionOfIsa(InstructionSet isa)
+{
+    switch (isa)
+    {
+        case InstructionSet_SSE:
+            return InstructionSet_SSE_X64;
+        case InstructionSet_SSE2:
+            return InstructionSet_SSE2_X64;
+        case InstructionSet_SSE41:
+            return InstructionSet_SSE41_X64;
+        case InstructionSet_SSE42:
+            return InstructionSet_SSE42_X64;
+        case InstructionSet_BMI1:
+            return InstructionSet_BMI1_X64;
+        case InstructionSet_BMI2:
+            return InstructionSet_BMI2_X64;
+        case InstructionSet_LZCNT:
+            return InstructionSet_LZCNT_X64;
+        case InstructionSet_POPCNT:
+            return InstructionSet_POPCNT_X64;
+        default:
+            unreached();
+            return InstructionSet_ILLEGAL;
+    }
+}
+
 //------------------------------------------------------------------------
 // lookupIsa: Gets the InstructionSet for a given class name
 //
 // Arguments:
 //    className -- The name of the class associated with the InstructionSet to lookup
+//    enclosingClassName -- The name of the enclosing class of X64 classes
 //
 // Return Value:
 //    The InstructionSet associated with className
-InstructionSet HWIntrinsicInfo::lookupIsa(const char* className)
+InstructionSet HWIntrinsicInfo::lookupIsa(const char* className, const char* enclosingClassName)
 {
     assert(className != nullptr);
 
-    if (className[0] == 'A')
+    if (strcmp(className, "X64") == 0)
+    {
+        assert(enclosingClassName != nullptr);
```

Comment:
Nit: I found the partially recursive calls like this hard to read. I would change the signature of lookupIsa to just `lookupIsa(const char* className)`, and the callsite to:

```
if (strcmp(className, "X64") == 0)
{
    isa = X64VersionOfIsa(lookupIsa(enclosingClassName));
}
else
{
    isa = lookupIsa(className);
}
```

Reply:
Thanks, done by splitting the code into two functions.